ModMenu/Helpers.cs

```diff
@@ -62,6 +67,77 @@ public void Register()
       }
     }
 
+
+    internal class SettingsDescriptionUpdater<T>
+        where T : SettingsEntityWithValueVM
+    {
+      private readonly string pathMainUi;
+      private readonly string pathDescriptionUi;
+
+      private Transform mainUI;
+      private Transform settingsUI;
+      private Transform descriptionUI;
+
+      private List<SettingsEntityWithValueView<T>> settingViews;
+      private SettingsDescriptionPCView descriptionView;
+
+      public SettingsDescriptionUpdater(string pathUI, string pathDesriptionUI)
+      {
+        pathMainUi = pathUI;
+        pathDescriptionUi = pathDesriptionUI;
+      }
+
+      private bool Ensure()
+      {
+        // UI tends to change frequently, ensure that eveything is up to date.
+
+        if ((mainUI = Game.Instance.RootUiContext.m_CommonView.transform) == null)
+          return false;
+
+        settingsUI = mainUI.Find(pathMainUi);
+        descriptionUI = mainUI.Find(pathDescriptionUi);
+        if (settingsUI == null || descriptionUI == null)
+          return false;
+
+        settingViews = settingsUI.gameObject.GetComponentsInChildren<SettingsEntityWithValueView<T>>().ToList();
+        descriptionView = descriptionUI.GetComponent<SettingsDescriptionPCView>();
+
+        if (settingViews == null || descriptionView == null || settingViews.Count == 0)
+          return false;
+
+        return true;
+      }
+
+      public bool TryUpdate(string title, string desription)
```

Comment:
Use xmldocs, make sure to call out the title restriction. Also spelling nit on `description`